frontend/src/components/editable-cell.tsx

```diff
@@ -81,6 +83,12 @@ export function EditableCell<T extends HasId, Field extends keyof T = keyof T>({
     }
 
     const formattedValue = isDate ? ("" + value).slice(0, 10) : value || "";
+    let displayedValue = isDate ? formatDate("" + value) : value;
```

Comment:
We're getting complicated here. Probably better to use a switch statement to handle all cases at once.